src/schemas/utilities/properties.ts

```diff
@@ -22,4 +28,103 @@ export function getSchemaPropertyLabel(property: SchemaProperty): string {
   const type = getSchemaPropertyTypeLabel(property.type)
 
   return property.title ?? property.format ?? type
+}
+
+export function getSchemaPropertyAllOfDefinitions(property: Require<SchemaProperty, 'anyOf'>, schema: Schema): SchemaProperty[] {
+  return property.anyOf.map(definition => {
+    if (isPropertyWith(definition, '$ref')) {
+      return getSchemaDefinition(schema, definition.$ref)
+    }
+
+    return definition
+  })
+}
+
+type InitialIndexContext = {
+  property: Require<SchemaProperty, 'anyOf'>,
+  value: SchemaValue,
+  schema: Schema,
+  api: CreateApi,
+}
+
+export async function getInitialIndexForSchemaPropertyAnyOfValue({ value, property, schema, api }: InitialIndexContext): Promise<number> {
+  // if there's no value default to showing the first definition
+  if (!isNotNullish(value)) {
+    return 0
+  }
+
+  const definitions = getSchemaPropertyAllOfDefinitions(property, schema)
+
+  // block documents are the only reason this utility is async.
+  // if the value is a block document reference we need to fetch the block document from the api
+  // to determine the block type
+  if (isBlockDocumentReferenceValue(value)) {
+    return await getBlockDocumentReferenceDefinitionIndex(value, definitions, api)
+  }
```

Comment:
The current implementation of this just grabs the first definition that is for a block type. But the example I was working off of is the dbt CLI Profile block which has a union of block types. This was the cleanest way I could think of the ensure the correct block definition was chosen. Shouldn't happen often so I'm fine with making an extra request here.